packages/gatsby-plugin-image/src/node-apis/image-processing.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  Node,
+  GatsbyCache,
+  Reporter,
+  ParentSpanPluginArgs,
+  Actions,
+} from "gatsby"
+import { fluid as fluidSharp, fixed as fixedSharp } from "gatsby-plugin-sharp"
+import { createFileNode } from "gatsby-source-filesystem/create-file-node"
+import fs from "fs-extra"
+import path from "path"
+import { ImageProps, SharpProps } from "../utils"
+import { getCustomSharpFields } from "./get-custom-sharp-fields"
+import { watchImage } from "./watcher"
+
+export interface IImageMetadata {
+  isFixed: boolean
+  contentDigest?: string
+  args: Record<string, unknown>
+  cacheFilename: string
+}
+
+export async function createImageNode({
+  fullPath,
+  createNodeId,
+  createNode,
+}: {
+  fullPath: string
+  createNodeId: ParentSpanPluginArgs["createNodeId"]
+  createNode: Actions["createNode"]
+}): Promise<Node | undefined> {
+  if (!fs.existsSync(fullPath)) {
+    return undefined
+  }
+  const file: Node = await createFileNode(fullPath, createNodeId, {})
+
+  if (!file) {
+    return undefined
+  }
+
+  file.internal.type = `StaticImage`
+
+  createNode(file)
+
+  return file
+}
+
+export async function writeImages({
+  images,
+  cacheDir,
+  reporter,
+  cache,
+  sourceDir,
+  createNodeId,
+  createNode,
+}: {
+  images: Map<string, ImageProps>
+  cacheDir: string
+  reporter: Reporter
+  cache: GatsbyCache
+  sourceDir: string
+  createNodeId: ParentSpanPluginArgs["createNodeId"]
+  createNode: Actions["createNode"]
+}): Promise<void> {
+  const promises = [...images.entries()].map(
+    async ([hash, { src, fluid, fixed, ...args }]) => {
+      // Default to fixed, but allow specifying either
+      const isFixed = !!(fixed ?? !fluid)
```

Comment:
This differs from our previous `isFixed` code. We should make them the same. 
```suggestion
      const isFixed = fixed ?? !fluid
```

Reply:
It needs casting to boolean, but I'll do it with `Boolean()` to make it clearer